### PR TITLE
hcloud discovery: Add testcase for key only labels

### PR DESCRIPTION
**Pull request by LKaemmerling**
This adds a test case for the hcloud discovery with key only labels.

Key Only Labels should already be possible as they are returned from the hcloud API `{"key":""}`

Fixes https://github.com/prometheus/prometheus/issues/9023

### CODE REVIEW

**Comment by roidelapluie** [comment]
This should be a new label, which is there for all the labels.

**Comment by roidelapluie** [comment]
(the new label should have  _labelpresent_ instead of _label_), see kubernetes_sd doc.

**Comment by LKaemmerling** [comment]
> (the new label should have _labelpresent_ instead of _label_), see kubernetes_sd doc.

@roidelapluie i changed it, thank you!

File: discovery/hetzner/hcloud.go [hunk 1]
```diff
@@ -125,6 +126,10 @@ func (d *hcloudDiscovery) refresh(ctx context.Context) ([]*targetgroup.Group, er
 		}
 		for labelKey, labelValue := range server.Labels {
 			label := model.LabelName(hetznerLabelHcloudLabel + strutil.SanitizeLabelName(labelKey))
+			if labelValue == "" {
+				label = model.LabelName(hetznerLabelHcloudLabelPresent + strutil.SanitizeLabelName(labelKey))
+				labelValue = "true"
+			}
```
Comment (roidelapluie) [inline discovery/hetzner/hcloud.go:132]: let's have this for every label, not just the empty ones.

File: docs/configuration/configuration.md [hunk 2]
```diff
@@ -1211,6 +1211,7 @@ The labels below are only available for targets with `role` set to `hcloud`:
 * `__meta_hetzner_hcloud_disk_size_gb`: the disk size of the server (in GB)
 * `__meta_hetzner_hcloud_private_ipv4_<networkname>`: the private ipv4 address of the server within a given network
 * `__meta_hetzner_hcloud_label_<labelname>`: each label of the server
+* `__meta_hetzner_hcloud_labelpresent_<labelname>`: `true` each label of the server that does not contain a value
```
Comment (roidelapluie) [inline docs/configuration/configuration.md:1214]:
```suggestion
* `__meta_hetzner_hcloud_labelpresent_<labelname>`: `true` each label of the server.
```

**Review by roidelapluie — COMMENTED** [review]
Thanks! It should be for every label.

**Comment by LKaemmerling** [comment]
@roidelapluie changed :)

File: discovery/hetzner/hcloud.go [hunk 3]
```diff
@@ -124,8 +125,12 @@ func (d *hcloudDiscovery) refresh(ctx context.Context) ([]*targetgroup.Group, er
 			}
 		}
 		for labelKey, labelValue := range server.Labels {
-			label := model.LabelName(hetznerLabelHcloudLabel + strutil.SanitizeLabelName(labelKey))
-			labels[label] = model.LabelValue(labelValue)
+			presentLabel := model.LabelName(hetznerLabelHcloudLabelPresent + strutil.SanitizeLabelName(labelKey))
+			labels[presentLabel] = model.LabelValue("true")
+			if labelValue == "" {
+				label := model.LabelName(hetznerLabelHcloudLabel + strutil.SanitizeLabelName(labelKey))
```
Comment (roidelapluie) [inline discovery/hetzner/hcloud.go:131]: this if should go away. We should have the label for every hcloud label.

File: docs/configuration/configuration.md [hunk 4]
```diff
@@ -1211,6 +1211,7 @@ The labels below are only available for targets with `role` set to `hcloud`:
 * `__meta_hetzner_hcloud_disk_size_gb`: the disk size of the server (in GB)
 * `__meta_hetzner_hcloud_private_ipv4_<networkname>`: the private ipv4 address of the server within a given network
 * `__meta_hetzner_hcloud_label_<labelname>`: each label of the server
+* `__meta_hetzner_hcloud_labelpresent_<labelname>`: `true` each label of the server
```
Comment (roidelapluie) [inline docs/configuration/configuration.md:1214]:
```suggestion
* `__meta_hetzner_hcloud_labelpresent_<labelname>`: `true` for each label of the server
```

File: discovery/hetzner/hcloud.go (hunk 3 above)
Comment (LKaemmerling) [reply discovery/hetzner/hcloud.go:131]: This doesn't make sense, why should the label be there if prometheus skips it at all?

Comment (roidelapluie) [reply discovery/hetzner/hcloud.go:131]: ooh it is the normal label :)

Any way works, I just prefer as less `if` as possible.

**Comment by roidelapluie** [comment]
I guess the tests should fail and you will need to rebase to fix the DCO.

Thanks!

**Review by roidelapluie — APPROVED** [review]
Last nits: Can you remove the if branch, and add another (filled) label in the tests?

**Comment by LKaemmerling** [comment]
@roidelapluie i added this case :)

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**